Extra features

TomSerrano opened the pull request.

TomSerrano committed:
Added additional features to repo_data

These features should probably be run per repo rather than per file, so it makes sense to throw them in with repository_data() here instead of making additional calls from app_interface. The order of the functions was rearranged slightly to compensate.

TomSerrano committed:
Added get_file_features()

TomSerrano committed:
Now assembles data sets with the additional features

TomSerrano committed:
Moved get_file_features() from app_interface to application

TomSerrano committed:
Moved get_file_features() from app_interface to application

Bonfire reviewed (changes requested):
Please review my feedback below

TomSerrano committed:
Fixed typo in get_file_features()

Bonfire reviewed (changes requested):
Please review my comments below.

TomSerrano committed:
Fixed fileLineCount and fileWordCount

TomSerrano committed:
Undid previous fix

TomSerrano committed:
get_file_features() now discards 'weird' files

It should delete from the record any files it can't successfully retrieve all the file features from.

Bonfire reviewed (changes requested):
Please review my feedback below.

Receiving the following error when supplying the following input.

Input:
```
python .\app_interface.py C:\Users\bonfi\Documents\GitHub\knowledge
```

Output:
```
Using TensorFlow backend.
Traceback (most recent call last):
  File ".\app_interface.py", line 122, in <module>
    app_interface()
  File ".\app_interface.py", line 114, in app_interface
    results = predict(final_data)
  File "C:\Users\bonfi\Documents\GitHub\SecurityWhale\Backend\predictor.py", line 22, in predict
    features = scaler.transform(features)
  File "C:\Users\bonfi\AppData\Roaming\Python\Python37\site-packages\sklearn\preprocessing\data.py", line 769, in transform
    X -= self.mean_
ValueError: operands could not be broadcast together with shapes (7,13) (31,) (7,13)
```

Bonfire commented:
Closing due to inactivity
